test/crontab/functional_test.exs

```diff
@@ -220,4 +220,54 @@ defmodule Crontab.FunctionalTest do
       assert Crontab.DateChecker.matches_date?(cron_expression, @start_date) == @matches_now
     end
   end
+
+  tests_find_date_tz_changes = [
+    # DST Shift with extra hour
+    {"30 2 * * *", [:earlier], "Europe/Zurich", ~N[2024-10-27 01:00:01],
+     ["2024-10-27T02:30:00+02:00", "2024-10-28T02:30:00+01:00"]},
+    {"30 2 * * *", [:later], "Europe/Zurich", ~N[2024-10-27 01:00:01],
+     ["2024-10-27T02:30:00+01:00", "2024-10-28T02:30:00+01:00"]},
+    {"30 2 * * *", [:earlier, :later], "Europe/Zurich", ~N[2024-10-27 01:00:01],
+     ["2024-10-27T02:30:00+02:00", "2024-10-27T02:30:00+01:00", "2024-10-28T02:30:00+01:00"]},
+    {"30 2 * * *", [], "Europe/Zurich", ~N[2024-10-27 01:00:01], ["2024-10-28T02:30:00+01:00"]},
+
+    # DST Shift with missing hour
+    {"30 2 * * *", [:earlier], "Europe/Zurich", ~N[2024-03-31 01:00:01],
+     ["2024-04-01T02:30:00+02:00"]},
+    {"30 2 * * *", [:later], "Europe/Zurich", ~N[2024-03-31 01:00:01],
+     ["2024-04-01T02:30:00+02:00"]},
+    {"30 2 * * *", [:earlier, :later], "Europe/Zurich", ~N[2024-03-31 01:00:01],
+     ["2024-04-01T02:30:00+02:00"]},
+    {"30 2 * * *", [], "Europe/Zurich", ~N[2024-03-31 01:00:01], ["2024-04-01T02:30:00+02:00"]},
+
+    # Half Hour Shift
+    {"*/30", [:earlier], "Australia/Lord_Howe", ~N[2024-04-07 01:15:01],
```

Comment:
As the cron expression expects the frequency as "every 30 minutes", `shift/4` ignores the ambiguity opts. These half hour shift tests are invalid.